### PR TITLE
feat: Do not marshal/unmarshal on the same node

### PR DESCRIPTION
**Pull request by hueypark**

### CODE REVIEW

File: lrdd/row.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+type MarshalUnmarshaler interface {
+	MarshalMsg([]byte) ([]byte, error)
+	UnmarshalMsg([]byte) ([]byte, error)
+	ID() RowID
+	String() string
+}
+
+type RowID int32
```
Comment (KimMachineGun) [inline lrdd/row.go:17]: `RowID` represents the type of row, not an identifier of each row itself, so I think `RowType` is more intuitive.

File: lrdd/row.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	default:
+		return rowTypes[rowID]()
+	}
+}
+
+func RegisterValue(rowID RowID, newFunc func() MarshalUnmarshaler) {
+	switch rowID {
+	case RowIDBytes:
+	case RowIDUint64:
+		panic("cannot register builtin row type")
```
Comment (KimMachineGun) [inline lrdd/row.go:40]: I think it would be clear to register builtin types in `rowTypes`, and unify the checking logic.

Comment (KimMachineGun) [reply lrdd/row.go:40]: To provide more kind error message, registering builtin types separately from the user-registered types and checking both maps is also fine.

**Review by KimMachineGun — CHANGES_REQUESTED** [review]

File: lrdd/row.go (hunk 1 above)
Comment (hueypark) [reply lrdd/row.go]: Done. 468c7ec22ea0f4d25e9ae093e441a53c8532350b

File: lrdd/row.go (hunk 2 above)
Comment (hueypark) [reply lrdd/row.go]: Done. 6fad2a5ed0d37b2e1d2f4565d44a846a5066eecc

**Pull request merged**